templates/etc/apt/sources.list.j2

```diff
@@ -27,7 +27,7 @@
 {%     endfor %}
 {%   else %}
 {%     for entry_uri in (entry.uri|d([]) + entry.uris|d([])) %}
-{%       if entry_uri in apt__tpl_seen_sources %}
+{%       if (entry_uri in apt__tpl_seen_sources and apt__sources_filter_duplicates|bool) %}
```

Comment:
@ypid Yes, that's basically what's currently done - `apt__*_sources` lists are parsed and a new unified list is generated in the template, and this list is used to generate the finished configuration file.